<?xml version="1.0"?>
<launch>

  <arg name="robot_name"              default="/" />
  <arg name="calibrate"               default="false" />
  <arg name="servo_calibration_file"  default="$(find fennee_control)/config/servos_calibration.yaml"/>
  <arg name="hardware_connected"      default="true" />
  <arg name="servo_max_angle"         default="180" />

  <group ns="$(arg robot_name)">
    <param name="tf_prefix"               value="$(arg robot_name)"/>
    <param name="servo_calibration_file"  value="$(arg servo_calibration_file)"/>
    <param name="hardware_connected"      value="$(arg hardware_connected)"/>
    <param name="servo_max_angle"         value="$(arg servo_max_angle)"/>

    <!-- ==================== LOAD PARAMETERS ==================== -->
    <rosparam command="load" ns="servo_calibration" file="$(arg servo_calibration_file)" />
    <rosparam file="$(find fennee_config)/config/ros_control/hardware.yaml" command="load"/>


    <node name="servo_interface" pkg="fennee_control" type="servo_interface.py" output="screen" />

    <node if="$(arg calibrate)" name="servo_calibration" pkg="fennee_control" type="servo_calibration.py" output="screen"  />
  </group>

</launch>
